pkg/corerp/datamodel/environment.go

```diff
@@ -123,3 +128,29 @@ type ProvidersAWS struct {
 	// Scope is the target level for deploying the aws resources
 	Scope string `json:"scope,omitempty"`
 }
+
+// EnvironmentVariables describes structure enabling environment variables to be set.
+type EnvironmentVariables struct {
+	// Additional properties for the resource
+	AdditionalProperties map[string]any `json:"additionalProperties,omitempty"`
+
+	// Secrets values provided for the resource
+	Secrets map[string]ProviderSecret `json:"secrets,omitempty"`
+}
+
+// ProviderSecret specifies the secret details for the provider.
+type ProviderSecret struct {
+	// Source represents the resource id for the secret store containing credentials.
+	Source string `json:"source,omitempty"`
+
+	// Key represents the key to the secret in the secret store.
+	Key string `json:"key,omitempty"`
+}
+
+type ProviderConfigProperties struct {
+	// Additional properties for the resource
```

Comment:
Let's update this comment as well. What is a "resource" here? What are the additional properties expected to be?